src/WebJobs.Script.WebHost/Formatting/PlaintextMediaTypeFormatter.cs

```diff
@@ -42,17 +42,17 @@ public override bool CanWriteType(Type type)
             }
         }
 
-        public override Task WriteToStreamAsync(Type type, object value, Stream writeStream, HttpContent content, TransportContext transportContext, CancellationToken cancellationToken)
+        public override async Task WriteToStreamAsync(Type type, object value, Stream writeStream, HttpContent content, TransportContext transportContext, CancellationToken cancellationToken)
         {
             if (value == null)
             {
-                return Task.CompletedTask;
+                return;
             }
 
             Encoding selectedEncoding = SelectCharacterEncoding(content.Headers);
             using (var writer = new StreamWriter(writeStream, selectedEncoding, bufferSize: 1024, leaveOpen: true))
             {
-                return writer.WriteAsync((string)value);
+                await writer.WriteAsync((string)value);
```

Comment:
This, technically, still returns an uncompleted task (as soon as await is hit), which is awaited on. The issue was completely on my side, and if you look closely, you'll see it (it smacks you right on the face when you notice).
The writer was disposed of before returning that (uncompleted) task, which means that if anything was larger than the buffer, the second pass would run into the problem.